ftplugin/markdown.vim

```diff
@@ -731,18 +731,18 @@ function! s:MarkdownClearSyntaxVariables()
 endfunction
 
 augroup Mkd
-    autocmd! * <buffer>
-    autocmd BufWinLeave <buffer> mkview!
-    autocmd BufWinEnter <buffer> silent! loadview
-    autocmd BufWinEnter <buffer> call s:MarkdownRefreshSyntax(1)
+    autocmd!
+    autocmd BufWinEnter *.md call s:MarkdownRefreshSyntax(1)
     " workaround, even without options in viewoptions it still saves this
     " so it needs to be reset every time so that someone who has set their own
     " foldmethod(not manual) isn't stuck on it.
-    autocmd BufWinEnter <buffer> setlocal foldmethod=manual
-    autocmd BufUnload <buffer> call s:MarkdownClearSyntaxVariables()
-    autocmd BufWritePost <buffer> call s:MarkdownRefreshSyntax(0)
-    autocmd InsertEnter,InsertLeave <buffer> call s:MarkdownRefreshSyntax(0)
-    autocmd CursorHold,CursorHoldI <buffer> call s:MarkdownRefreshSyntax(0)
+    autocmd BufWinEnter *.md setlocal foldmethod=manual
+    autocmd BufUnload *.md call s:MarkdownClearSyntaxVariables()
+    autocmd BufWritePost *.md call s:MarkdownRefreshSyntax(0)
+    autocmd InsertEnter,InsertLeave *.md call s:MarkdownRefreshSyntax(0)
+    autocmd CursorHold,CursorHoldI *.md call s:MarkdownRefreshSyntax(0)
+
+    autocmd TextYankPost *.md call v:lua.require("markdown").reenum_wrapper(v:true)
```

Comment:
There's no event, that I know of, that is triggered exclusively for deletion. As such, in `reenum_wrapper` I have checked if the operator is yank. Currently, we are re-enumerating the list upon yanking, but it should be removed. The reason I kept it is to manually trigger a re-enumeration in case the ordering is messed up.